src/map.h

```diff
@@ -1052,6 +1067,20 @@ void add_corpse(int x, int y);
          * tr_null trap.
          */
         std::vector< std::vector<tripoint> > traplocs;
+
+    // cached values for map visibility calculations
+    int g_light_level;
```

Comment:
here's where it was already being done: https://github.com/CleverRaven/Cataclysm-DDA/pull/11797/files#diff-6f8417614661f6a364e536794cfa57deL1430